roles/control_center_next_gen/tasks/main.yml

```diff
@@ -449,28 +449,59 @@
     src: "{{ logredactor_rule_path_local }}"
     dest: "{{ logredactor_rule_path }}"
     mode: '640'
-  when: (control_center_next_gen_custom_log4j|bool) and (logredactor_enabled|bool) and (logredactor_rule_url == '')
+  when: (control_center_next_gen_custom_log4j2|bool) and (logredactor_enabled|bool) and (logredactor_rule_url == '')
   tags:
     - log
 
-- name: Configure logredactor
-  include_role:
-    name: common
-    tasks_from: configure_logredactor.yml
-  vars:
-    log4j_file: "{{control_center_next_gen.log4j_file}}"
-    logredactor_rule_var: "{% if logredactor_rule_url == '' %}{{logredactor_rule_path}}{% else %}{{logredactor_rule_url}}{% endif %}"
-    logredactor_policy_refresh_interval_var: "{{logredactor_policy_refresh_interval}}"
-  with_items: "{{ control_center_next_gen_logredactor_logger_specs_list }}"
-  when: (control_center_next_gen_custom_log4j|bool) and (logredactor_enabled|bool)
+- name: Check if Control Center Next Gen log4j file exists
+  stat:
+    path: "{{ control_center_next_gen.log4j_file }}"
+  register: log4j_file_stat
+  tags:
+    - log
+
+- name: Backup Control Center Next Gen log4j file if it exists
+  copy:
+    src: "{{ control_center_next_gen.log4j_file }}"
+    dest: "{{ control_center_next_gen.log4j_file }}.bak"
+    remote_src: yes
+  when: log4j_file_stat.stat.exists
+  tags:
+    - log
+
+- name: Update log4j Config for Log Cleanup
+  confluent.platform.update_log4j2:
+    path: "{{control_center_next_gen.log4j_file}}"
+    max: "{{control_center_next_gen_max_log_files}}"
+    size: "{{control_center_next_gen_log_file_size}}"
+    root_level: "{{control_center_next_gen_log4j2_root_logger_level}}"
+    root_appenders: "{{ control_center_next_gen_log4j2_root_appenders }}"
+    add_redactor: "{{ logredactor_enabled|bool }}"
+    redactor_refs: "{{ control_center_next_gen_logredactor_logger_specs_list | map(attribute='appenderRefs') | list }}"
+    redactor_rules: "{{ (logredactor_rule_url == '') | ternary(logredactor_rule_path, logredactor_rule_url) }}"
```

Comment:
needs to be updated like https://github.com/confluentinc/cp-ansible/blob/8.0.x/roles/kafka_broker/tasks/main.yml#L373